Jenkinsfile

```diff
@@ -88,54 +88,80 @@ def publishChart(imageName) {
-        deployPR(kubeCredsId, registry, imageName, containerTag, extraCommands)
-        echo "Build available for review"
+  try {
+    stage('Set branch, PR, and containerTag variables') {
+      (branch, pr, containerTag, mergedPrNo) = getVariables(repoName)
+      if (pr) {
+        sh "echo Building $pr"
+      } else if (branch == "master") {
+        sh "echo Building master branch"
+      } else {
+        currentBuild.result = 'ABORTED'
+        error('Build aborted - not a PR or a master branch')
       }
+      updateGithubCommitStatus('Build started','PENDING')
+    }    
+    stage('Build test image') {
+      buildTestImage(imageName, BUILD_NUMBER)
     }
-  }
-  if (pr == '') {
-    stage('Publish chart') {
-      publishChart(imageName)
+    stage('Run tests') {
+      runTests(imageName, BUILD_NUMBER)
     }
-  }
-  if (mergedPrNo != '') {
-    stage('Remove merged PR') {
-      sh "echo removing deployment for PR $mergedPrNo"
-      undeployPR(kubeCredsId, imageName, mergedPrNo)
+    // note: there should be a `build production image` step here,
+    // but the docker file is currently not set up to create a production only image
+    stage('Push container image') {
+      pushContainerImage(registry, regCredsId, imageName, containerTag)
+    }
+    if (pr != '') {
+      stage('Helm install') {
+        withCredentials([
+            string(credentialsId: 'messageQueueHostPR', variable: 'messageQueueHost'),
+            usernamePassword(credentialsId: 'scheduleListenPR', usernameVariable: 'scheduleQueueUsername', passwordVariable: 'scheduleQueuePassword'),
+            usernamePassword(credentialsId: 'paymentListenPR', usernameVariable: 'paymentQueueUsername', passwordVariable: 'paymentQueuePassword'),
+            string(credentialsId: 'postgresExternalNamePaymentsPR', variable: 'postgresExternalName'),
+            usernamePassword(credentialsId: 'postgresPaymentsPR', usernameVariable: 'postgresUsername', passwordVariable: 'postgresPassword'),
+          ]) {
+          def extraCommands = "--values ./helm/ffc-demo-payment-service/jenkins-aws.yaml --set container.messageQueueHost=\"$messageQueueHost\",container.scheduleQueueUser=\"$scheduleQueueUsername\",container.scheduleQueuePassword=\"$scheduleQueuePassword\",container.paymentQueueUser=\"$paymentQueueUsername\",container.paymentQueuePassword=\"$paymentQueuePassword\",postgresExternalName=\"$postgresExternalName\",postgresUsername=\"$postgresUsername\",postgresPassword=\"$postgresPassword\""
+          deployPR(kubeCredsId, registry, imageName, containerTag, extraCommands)
+          echo "Build available for review"
+        }
+      }
+      if (pr == '') {
+        stage('Publish chart') {
+          publishChart(imageName)
+        }
+      }
+      if (mergedPrNo != '') {
+        stage('Remove merged PR') {
+          sh "echo removing deployment for PR $mergedPrNo"
+          undeployPR(kubeCredsId, imageName, mergedPrNo)
+        }
+      }
```

Comment:
Other than the general changes mentioned in the slack channel, you're missing the final 
`updateGithubCommitStatus('Build successful', 'SUCCESS', repoUrl, commitSha)`

Reply:
Added missing call

Reply:
Also updated variables